Don't recommend WCS&T if Woo Shipping or Woo Tax is active

waclawjacek opened the pull request:
<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

If either WC Shipping or WC Tax is active, don't recommend installing WCS&T in the following places:

- **Edit order page**
- **WC > Home > Task list > Collect sales tax:** Do not recommend any tax extensions.
- **WC > Home > Task list > Select your shipping options:** Do not recommend any shipping extensions.
- **WC > Settings > Shipping**
- **Activity Panel help**
- **WC Core Profiler (setup wizard):** Do not display WCS&T items.

<!-- Begin testing instructions -->


<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

If either the upcoming Woo Shipping or Woo Tax are installed and active, don't suggest WooCommerce Shipping & Tax.

</details>

Commit:
No WCS&T shipping banner with Woo Shipping or Woo Tax present

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

kloon reviewed (commented):
@waclawjacek I believe there are other places where we would also want to not display the banners.

https://github.com/woocommerce/woocommerce/blob/8ee74ede0ebc1957089af08207ae2a1a8250610e/plugins/woocommerce/includes/admin/class-wc-admin-addons.php#L523

https://github.com/woocommerce/woocommerce/blob/8ee74ede0ebc1957089af08207ae2a1a8250610e/plugins/woocommerce-admin/client/activity-panel/panels/help.js#L191-L213

https://github.com/woocommerce/woocommerce/blob/08344736db1446fc6e2c2a3a4803f639c9c0ab6e/plugins/woocommerce-admin/client/shipping/experimental-shipping-recommendations.tsx#L52-L57

https://github.com/woocommerce/woocommerce/blob/8ee74ede0ebc1957089af08207ae2a1a8250610e/plugins/woocommerce-admin/client/shipping/shipping-recommendations.tsx#L106-L108

These are just a couple of instances where I found recommending WCS&T.

I wonder if we cannot get @woocommerce/ghidorah to assist us identifying and updating/disabling the WCS&T onboarding tasks as it seems there are checks scattered around in a lot of places in the codebase.

Commit:
Merge branch 'trunk' into tweak/no-wcservices-banner-with-woo-shipping-or-tax-present

waclawjacek commented:
@kloon Good point! I see some of these are in deprecated code that is no longer called anywhere but we could use this PR (or spin up new ones?) to add conditions to the remaining prompts to install WCS&T.

I started looking into the other places but testing isn't trivial so it's taking a bit longer.

ilyasfoo commented:
Hi there! I've looked into WCS in onboarding related areas, we have two areas where we promote WCS:

### Setup wizard

[Screenshot](https://github.com/woocommerce/woocommerce/assets/3747241/c940076b-dd26-4372-8535-558c727e1178)

1. Set up a fresh store, go through setup wizard and select a US state as the store location and stop at the plugins page
2. Observe the WC&T promotion in the plugins list

This list is rendered via a JSON [list here](https://github.com/woocommerce/woocommerce/blob/2b4b8be4e669131028ee650aa0c470364c8e0b7f/plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/DefaultFreeExtensions.php#L70-L75) under `obw/core-profiler` key, I think we can simply add a rule to ensure Shipping and Tax plugin is not activated. For example, [this logic](https://github.com/woocommerce/woocommerce/blob/2b4b8be4e669131028ee650aa0c470364c8e0b7f/plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/DefaultFreeExtensions.php#L407-L415) checks `woocommerce-services` plugin so it's only shown when it's not activated. We'll need to apply the rule to both `woocommerce-services:shipping` and `woocommerce-services:tax`.

There's two list that we'll need to update: [core](https://github.com/woocommerce/woocommerce/blob/2b4b8be4e669131028ee650aa0c470364c8e0b7f/plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/DefaultFreeExtensions.php), [wccom](https://github.com/Automattic/woocommerce.com/blob/247aa4ddef054297bdadb0bb7eb5d57566ca82cc/plugins/wccom-plugins/obw-free-extensions/includes/v4/extensions-v4.php). Note that the wccom list is a remote endpoint list, this particular API is used by WooCommerce version since 8.8.



### Shipping recommendation

[Screenshot](https://github.com/woocommerce/woocommerce/assets/3747241/f04df722-be38-42c8-a88b-bc6d9cc7bed9)

1. Set up a fresh store, go through setup wizard and select a US state as the store location
4. Click on `Review shipping options`

The simplest way to disable it is to add another condition for `showShippingRecommendationsStep` [here](https://github.com/woocommerce/woocommerce/blob/2b4b8be4e669131028ee650aa0c470364c8e0b7f/plugins/woocommerce-admin/client/shipping/experimental-shipping-recommendations.tsx#L65) to ensure `true` is passed only when both Shipping and Tax plugins do not exist.

Sidenote, I noticed a bug when clicking on `Get Started` which leads to a blank screen. I'll look into that further.

waclawjacek commented:
Marking as blocked until namespaces are resolved for the shipping and tax extension loaders which the code in this PR uses:

- https://github.com/woocommerce/woocommerce-shipping/pull/484
- https://github.com/woocommerce/woocommerce-tax/pull/79

Commit:
Update class references to current namespaces

Commit:
Merge branch 'trunk' into tweak/no-wcservices-banner-with-woo-shipping-or-tax-present

Commit:
Only allow viewing ExperimentalShippingRecommendation without WCShip active

Commit:
Never recommend WCS&T if WCShipping or WCTax is present

Commit:
Add changefile(s) from automation for the following project(s): @woocommerce/data, woocommerce

Commit:
Remove WCS&T if WCShip is present from free extension and shipping rec lists

Commit:
Merge branch 'tweak/no-wcservices-banner-with-woo-shipping-or-tax-present' of github.com:woocommerce/woocommerce into tweak/no-wcservices-banner-with-woo-shipping-or-tax-present

waclawjacek commented on plugins/woocommerce/src/Internal/Admin/ShippingLabelBanner.php:
```diff
@@ -60,7 +60,9 @@ private function should_show_meta_box() {
 				class_exists( '\WC_Integration_ShippingEasy' ) ||
-				class_exists( '\WC_ShipStation_Integration' );
+				class_exists( '\WC_ShipStation_Integration' ) ||
+				class_exists( '\Automattic\WCShipping\Loader' ) ||
+				class_exists( '\Automattic\WCTax\Loader' );
```
This prevents the following banner from being displayed on the Edit order page when either WC Shipping or WC Tax are present:

![image](https://github.com/woocommerce/woocommerce/assets/1759681/340d07e8-ac42-4151-8432-bc68a41fb88a)

waclawjacek commented on plugins/woocommerce-admin/client/shipping/experimental-shipping-recommendations.tsx:
```diff
@@ -49,6 +49,10 @@ const ShippingRecommendations: React.FC = () => {
 		};
 	} );
 
+	if ( activePlugins.includes( 'woocommerce-shipping' ) ) {
+		return <ShippingTour showShippingRecommendationsStep={ false } />;
+	}
+
```
This prevents this banner from being displayed on the WC > Settings > Shipping page if WC Shipping is active:

<img width="722" alt="Screenshot 2024-06-06 at 13 31 02" src="https://github.com/woocommerce/woocommerce/assets/1759681/177f8ff6-7957-4368-a608-195f0d9718de">

The banner mentions "WooCommerce Shipping" but what it installs is the WCS&T extension which we don't want to recommend if WC Shipping is active.

waclawjacek commented on plugins/woocommerce-admin/client/shipping/shipping-recommendations.tsx:
```diff
@@ -103,7 +103,10 @@ const ShippingRecommendations: React.FC = () => {
 	);
 
-	if ( activePlugins.includes( 'woocommerce-services' ) ) {
+	if (
+		activePlugins.includes( 'woocommerce-services' ) ||
+		activePlugins.includes( 'woocommerce-shipping' )
+	) {
```
Practically the same things [as here](https://github.com/woocommerce/woocommerce/pull/46929/files#r1629347886) - same conditions and idea. The difference is that area on the WC > Settings > Shipping page will sometimes display this banner here, and sometimes the one in that other place. This depends on the `shipping-smart-defaults` feature flag.

Long story short, different code governing a banner that looks almost the same but we need to disable both, hence the code here.

<img width="720" alt="Screenshot 2024-06-06 at 13 42 54" src="https://github.com/woocommerce/woocommerce/assets/1759681/3af1c0dc-c194-4c43-8989-fd7cfc648194">

Commit:
Remove unnecessary `visible` condition from shipping fill

waclawjacek commented on plugins/woocommerce-admin/client/activity-panel/panels/help.js:
```diff
@@ -190,7 +190,9 @@ function getProductsItems() {
 	const showWCS =
 		countryCode === 'US' &&
-		! activePlugins.includes( 'woocommerce-services' );
+		! activePlugins.includes( 'woocommerce-services' ) &&
+		! activePlugins.includes( 'woocommerce-shipping' );
```
Prevents WCS&T-related setup instructions from being displayed in the help sidebar when viewing shipping-related onboarding tasks with the WC Shipping plugin active:

<img width="742" alt="Screenshot 2024-06-06 at 13 54 06" src="https://github.com/woocommerce/woocommerce/assets/1759681/3f6a48b9-787a-4c02-a23b-ed65c944e7a2">

Note that despite the help item saying "WooCommerce Shipping", it actually refers to WCS&T's shipping functionality.

waclawjacek commented on plugins/woocommerce-admin/client/activity-panel/panels/help.js:
```diff
@@ -235,10 +237,15 @@ function getTaxItems( props ) {
 
 	const showWCS =
 		! taxJarActivated && // WCS integration doesn't work with the official TaxJar plugin.
+		! woocommerceTaxActivated &&
```
Prevents WCS&T-related setup instructions from being displayed in the help sidebar when viewing the "Collect sales tax" onboarding task if the WC Shipping plugin or the WC Tax plugin is active:

<img width="744" alt="Screenshot 2024-06-06 at 13 56 19" src="https://github.com/woocommerce/woocommerce/assets/1759681/4efb3b43-f154-4c9d-b480-72b987841d97">

Note that despite the help item saying "WooCommerce Tax", it actually refers to WCS&T's tax functionality.

waclawjacek commented on plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php:
```diff
@@ -129,6 +129,7 @@ public function get_additional_data() {
 		return array(
 			'avalara_activated'         => PluginsHelper::is_plugin_active( 'woocommerce-avatax' ),
 			'tax_jar_activated'         => class_exists( 'WC_Taxjar' ),
+			'woocommerce_tax_activated' => PluginsHelper::is_plugin_active( 'woocommerce-tax' ),
```
Informs [this functionality](https://github.com/woocommerce/woocommerce/pull/46929/files#r1629385636) whether the WC Tax plugin is installed.

waclawjacek commented on plugins/woocommerce-admin/client/task-lists/fills/tax/index.tsx:
```diff
@@ -162,6 +163,7 @@ const Tax: React.FC< TaxProps > = ( { onComplete, query, task } ) => {
 				component: WooCommerceTax,
 				isVisible:
 					! taxJarActivated && // WCS integration doesn't work with the official TaxJar plugin.
+					! woocommerceTaxActivated &&
```
Makes the WC > Home > task list > Collect sales tax task display this:

<img width="742" alt="Screenshot 2024-06-06 at 14 00 11" src="https://github.com/woocommerce/woocommerce/assets/1759681/5ff7489e-843f-4bc9-a9d4-495ac809f41e">

(The "Configure" button leads to the WC > Settings > Tax > Standard rates page.)

Instead of this:

<img width="740" alt="Screenshot 2024-06-06 at 13 59 50" src="https://github.com/woocommerce/woocommerce/assets/1759681/0132a9e6-2ca5-4c72-b364-51c1909439d5">

waclawjacek commented on plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/ExperimentalShippingRecommendation.php:
```diff
@@ -62,7 +62,7 @@ public function is_complete() {
 	 */
 	public function can_view() {
-		return Features::is_enabled( 'shipping-smart-defaults' );
+		return Features::is_enabled( 'shipping-smart-defaults' ) && ! PluginsHelper::is_plugin_active( 'woocommerce-shipping' );
```
Prevents access to the `ExperimentalShippingRecommendation` task that is displayed when the "Activate" button [in the banner on the WC > Settings > Shipping page](https://github.com/woocommerce/woocommerce/pull/46929/files#r1629347886) is clicked:

<img width="778" alt="Screenshot 2024-06-06 at 14 32 35" src="https://github.com/woocommerce/woocommerce/assets/1759681/dadba0db-9b4a-4e03-a564-cb736da13f20">

Not necessary because the banner will not be displayed with WC Shipping active but adding this just in case.

waclawjacek commented on plugins/woocommerce/src/Admin/Features/ShippingPartnerSuggestions/DefaultShippingPartners.php:
```diff
@@ -285,6 +285,15 @@ public static function get_all() {
 				'learn_more_link'   => 'https://woocommerce.com/products/shipping/',
 				'is_visible'        => array(
 					self::get_rules_for_countries( array( 'US' ) ),
+					array(
+						'type' => 'not',
+						'operand' => array(
+							array(
+								'type'    => 'plugins_activated',
+								'plugins' => array( 'woocommerce-shipping' ),
+							),
+						),
+					),
```
Controls what shipping extensions are recommended in the "Select your shipping options" WC Home task:

Before (suggesting WCS&T):

<img width="691" alt="Screenshot 2024-06-07 at 00 28 46" src="https://github.com/woocommerce/woocommerce/assets/1759681/fc7f4c8b-f98f-4600-a39a-ebb5804ddbb0">

After (no WCS&T suggestion):

<img width="696" alt="Screenshot 2024-06-07 at 00 29 04" src="https://github.com/woocommerce/woocommerce/assets/1759681/8b80cd6b-0ebf-4119-b69b-a0630a72a5f2">

Commit:
Cast DefaultShippingPartners rules to object

waclawjacek commented on plugins/woocommerce/src/Admin/Features/ShippingPartnerSuggestions/DefaultShippingPartners.php:
```diff
@@ -285,6 +285,15 @@ public static function get_all() {
 				'learn_more_link'   => 'https://woocommerce.com/products/shipping/',
 				'is_visible'        => array(
 					self::get_rules_for_countries( array( 'US' ) ),
+					(object) array(
+						'type' => 'not',
+						'operand' => array(
+							(object) array(
+								'type'    => 'plugins_activated',
+								'plugins' => array( 'woocommerce-shipping' ),
+							),
+						),
+					),
```
Controls what shipping extensions are recommended in the "Select your shipping options" WC Home task.

This file provides a fallback for when WC Core cannot download the most up-to-date recommendations from WCCOM. These recommendations are added to WCCOM by this PR: https://github.com/Automattic/woocommerce.com/pull/20789 and the changes here reproduce the logic added to WCCOM.

Before (suggesting WCS&T):

<img width="691" alt="Screenshot 2024-06-07 at 00 28 46" src="https://github.com/woocommerce/woocommerce/assets/1759681/fc7f4c8b-f98f-4600-a39a-ebb5804ddbb0">

After (no WCS&T suggestion):

<img width="696" alt="Screenshot 2024-06-07 at 00 29 04" src="https://github.com/woocommerce/woocommerce/assets/1759681/8b80cd6b-0ebf-4119-b69b-a0630a72a5f2">

Commit:
Update conditions to be context-agnostic

Commit:
Add missing type

Commit:
Add missing check for WCShip in tax/index.tsx

waclawjacek commented on plugins/woocommerce-admin/client/activity-panel/panels/help.js:
```diff
@@ -190,7 +190,10 @@ function getProductsItems() {
 	const showWCS =
 		countryCode === 'US' &&
-		! activePlugins.includes( 'woocommerce-services' );
+		! activePlugins.includes( 'woocommerce-services' ) &&
+		! activePlugins.includes( 'woocommerce-shipping' ) &&
+		! activePlugins.includes( 'woocommerce-tax' );
```
Prevents WCS&T-related setup instructions from being displayed in the help sidebar when viewing the "Select your shipping options" onboarding task with the WC Shipping plugin or the WC Tax plugin active:

<img width="742" alt="Screenshot 2024-06-06 at 13 54 06" src="https://github.com/woocommerce/woocommerce/assets/1759681/3f6a48b9-787a-4c02-a23b-ed65c944e7a2">

Note that despite the help item saying "WooCommerce Shipping", it actually refers to WCS&T's shipping functionality.

waclawjacek commented on plugins/woocommerce-admin/client/shipping/experimental-shipping-recommendations.tsx:
```diff
@@ -49,6 +49,13 @@ const ShippingRecommendations: React.FC = () => {
+		activePlugins.includes( 'woocommerce-shipping' ) ||
+		activePlugins.includes( 'woocommerce-tax' )
+	) {
+		return <ShippingTour showShippingRecommendationsStep={ false } />;
```
This prevents this banner from being displayed on the WC > Settings > Shipping page if WC Shipping or WC Tax is active:

<img width="722" alt="Screenshot 2024-06-06 at 13 31 02" src="https://github.com/woocommerce/woocommerce/assets/1759681/177f8ff6-7957-4368-a608-195f0d9718de">

The banner mentions "WooCommerce Shipping" but what it installs is the WCS&T extension which we don't want to recommend if WC Shipping is active.

waclawjacek commented on plugins/woocommerce-admin/client/shipping/shipping-recommendations.tsx:
```diff
@@ -103,7 +103,11 @@ const ShippingRecommendations: React.FC = () => {
+		activePlugins.includes( 'woocommerce-services' ) ||
+		activePlugins.includes( 'woocommerce-shipping' ) ||
+		activePlugins.includes( 'woocommerce-tax' )
+	) {
```
Practically the same things [as here](https://github.com/woocommerce/woocommerce/pull/46929/files#r1629347886) - same conditions and idea. The difference is that the banner and the task you will see is slightly different. Which banner you see depends on the `shipping-smart-defaults` feature flag.

<img width="720" alt="Screenshot 2024-06-06 at 13 42 54" src="https://github.com/woocommerce/woocommerce/assets/1759681/3af1c0dc-c194-4c43-8989-fd7cfc648194">

waclawjacek commented on plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/DefaultFreeExtensions.php:
```diff
@@ -413,6 +413,24 @@ public static function get_plugin( $slug ) {
 							),
 						),
 					),
+					array(
+						'type'    => 'not',
+						'operand' => array(
+							array(
+								'type'    => 'plugins_activated',
+								'plugins' => array( 'woocommerce-shipping' ),
+							),
+						),
+					),
+					array(
+						'type'    => 'not',
+						'operand' => array(
+							array(
+								'type'    => 'plugins_activated',
+								'plugins' => array( 'woocommerce-tax' ),
+							),
+						),
+					),
```
Controls presence of WCS&T branded as "WooCommerce Shipping" in the core profiler/setup wizard.

This file provides a fallback for when WC Core cannot download the most up-to-date recommendations from WCCOM. These recommendations are added to WCCOM by this PR: https://github.com/Automattic/woocommerce.com/pull/20789 and the changes here reproduce the logic added to WCCOM.

waclawjacek commented on plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/DefaultFreeExtensions.php:
```diff
@@ -537,6 +555,24 @@ public static function get_plugin( $slug ) {
 							),
 						),
 					),
+					array(
+						'type'    => 'not',
+						'operand' => array(
+							array(
+								'type'    => 'plugins_activated',
+								'plugins' => array( 'woocommerce-shipping' ),
+							),
+						),
+					),
+					array(
+						'type'    => 'not',
+						'operand' => array(
+							array(
+								'type'    => 'plugins_activated',
+								'plugins' => array( 'woocommerce-tax' ),
+							),
+						),
+					),
```
Controls presence of WCS&T branded as "WooCommerce Tax" in the core profiler/setup wizard.

waclawjacek commented on plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php:
```diff
@@ -127,9 +127,11 @@ public function is_complete() {
-			'woocommerce_tax_countries' => self::get_automated_support_countries(),
+			'avalara_activated'              => PluginsHelper::is_plugin_active( 'woocommerce-avatax' ),
+			'tax_jar_activated'              => class_exists( 'WC_Taxjar' ),
+			'woocommerce_tax_activated'      => PluginsHelper::is_plugin_active( 'woocommerce-tax' ),
+			'woocommerce_shipping_activated' => PluginsHelper::is_plugin_active( 'woocommerce-shipping' ),
```
Informs [the "Collect sales tax" task](https://github.com/woocommerce/woocommerce/pull/46929/files#r1629393692), [the "Select your shipping options" task](https://github.com/woocommerce/woocommerce/pull/46929/files#r1630344531), and [the help sidebar](https://github.com/woocommerce/woocommerce/pull/46929/files#r1629385636) whether the WC Shipping plugin and the WC Tax plugin are installed.

Commit:
Add tests for help.js. Extract existing tests to functions

Commit:
Tests for experimental-shipping-recommendations.tsx

Commit:
Add tests for shipping-recommendations.tsx

Commit:
Tests for client/task-lists/fills/tax/index.tsx

Commit:
Add ExperimentalShippingRecommendationTest

Commit:
Merge branch 'trunk' into tweak/no-wcservices-banner-with-woo-shipping-or-tax-present

Commit:
Update DefaultFreeExtensions

- Add rules for WCS&T tax feature in core profiler.
- Extract WCS&T tax countries to method.
- Adjust logic in $remove_plugins_activated_rule.

Commit:
Undo earlier update to $remove_plugins_activated_rule

waclawjacek commented on plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/DefaultFreeExtensions.php:
```diff
@@ -468,63 +486,22 @@ public static function get_plugin( $slug ) {
 					'</a>'
 				),
 				'is_visible'     => array(
+					self::get_rules_for_wcservices_tax_countries(),
```
Controls presence of WCS&T branded as "WooCommerce Tax" in the core profiler/setup wizard.

This file provides a fallback for when WC Core cannot download the most up-to-date recommendations from WCCOM. These recommendations are added to WCCOM by this PR: Automattic/woocommerce.com#20789 and the changes here reproduce the logic added to WCCOM.

waclawjacek commented on plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/ExperimentalShippingRecommendation.php:
```diff
@@ -62,7 +62,9 @@ public function is_complete() {
 	 */
 	public function can_view() {
-		return Features::is_enabled( 'shipping-smart-defaults' );
+		return Features::is_enabled( 'shipping-smart-defaults' ) &&
+		       ! PluginsHelper::is_plugin_active( 'woocommerce-shipping' ) &&
+		       ! PluginsHelper::is_plugin_active( 'woocommerce-tax' );
```
Prevents access to the `ExperimentalShippingRecommendation` task that is displayed when the "Activate" button [in the banner on the WC > Settings > Shipping page](https://github.com/woocommerce/woocommerce/pull/46929/files#r1629347886) is clicked:

<img width="778" alt="Screenshot 2024-06-06 at 14 32 35" src="https://github.com/woocommerce/woocommerce/assets/1759681/dadba0db-9b4a-4e03-a564-cb736da13f20">

Not necessary because the banner will not be displayed with WC Shipping active but adding this just in case.

Note that despite the screen saying "WooCommerce Shipping", it actually refers to WCS&T.

waclawjacek commented on plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/DefaultFreeExtensions.php:
```diff
@@ -468,63 +486,22 @@ public static function get_plugin( $slug ) {
-							array(
-								'type'      => 'base_location_country',
-								'value'     => 'FR',
-								'operation' => '=',
-							),
-							array(
-								'type'      => 'base_location_country',
-								'value'     => 'GB',
-								'operation' => '=',
-							),
-							array(
-								'type'      => 'base_location_country',
-								'value'     => 'DE',
-								'operation' => '=',
-							),
-							array(
-								'type'      => 'base_location_country',
-								'value'     => 'CA',
-								'operation' => '=',
-							),
-							array(
-								'type'      => 'base_location_country',
-								'value'     => 'AU',
-								'operation' => '=',
-							),
-							array(
-								'type'      => 'base_location_country',
-								'value'     => 'GR',
-								'operation' => '=',
-							),
-							array(
-								'type'      => 'base_location_country',
-								'value'     => 'BE',
-								'operation' => '=',
-							),
-							array(
-								'type'      => 'base_location_country',
-								'value'     => 'PT',
-								'operation' => '=',
-							),
```
Not really removed, just extracted to a new method `get_rules_for_wcservices_tax_countries()`.

Commit:
Merge branch 'trunk' into tweak/no-wcservices-banner-with-woo-shipping-or-tax-present

waclawjacek commented:
I am closing this PR and opening individual PRs for the changes for ease of reviewing.

You can see the list of PRs that supersede this one in this issue: https://github.com/woocommerce/woocommerce-shipping/issues/187
